tests/workflows/update/update.go

```diff
@@ -0,0 +1,44 @@
+package update
+
+import (
+	"fmt"
+
+	"go.temporal.io/sdk/workflow"
+)
+
+const (
+	FetchAndAdd = "fetch_and_add"
+	Done        = "done"
+)
+
+func Counter(ctx workflow.Context, val int) (int, error) {
+	log := workflow.GetLogger(ctx)
+	counter := val
+
+	if err := workflow.SetUpdateHandlerWithOptions(
+		ctx,
+		FetchAndAdd,
+		func(ctx workflow.Context, i int) (int, error) {
+			tmp := counter
+			counter += i
+			log.Info("counter updated", "added", i, "new-value", counter)
+			return tmp, nil
+		},
+		workflow.UpdateHandlerOptions{Validator: nonNegative},
+	); err != nil {
+		return 0, err
+	}
+
+	_ = workflow.GetSignalChannel(ctx, Done).Receive(ctx, nil)
+	return counter, ctx.Err()
```

Comment:
This isn't the right way to handle context errors. It should be this instead:

```suggestion
	if err := workflow.GetSignalChannel(ctx, Done).Receive(ctx, nil); err != nil {
	    return 0, err
	}
	return counter, nil
```

If the context is canceled, then GetSignalChannel will return that to you as an error (unless the implementation is wrong)

Reply:
nice, TIL